src/node_zlib.cc

```diff
@@ -1286,6 +1286,35 @@ struct MakeClass {
   }
 };
 
+template <typename T>
+T CallOnSequence(v8::Isolate* isolate,
+                 Local<Value> value,
+                 std::function<T(const char* data, size_t size)> callback) {
+  if (value->IsString()) {
+    Utf8Value data(isolate, value);
+    return callback(data.out(), data.length());
+  } else {
+    ArrayBufferViewContents<char> data(value);
+    return callback(data.data(), data.length());
+  }
+}
+
+// TODO(joyeecheung): use fast API
+static void CRC32(const FunctionCallbackInfo<Value>& args) {
+  CHECK(args[0]->IsArrayBufferView() || args[0]->IsString());
```

Comment:
Something like CHECK_GE(args.length(), 1) is missing

Reply:
If it's empty, this would simply fail (the [] operator returns undefined if it's out of bounds).